magma/ref.py

```diff
@@ -207,8 +208,17 @@ def qualifiedname(self, sep="."):
     def anon(self):
         return self.view.port.anon()
 
-    def root(self):
+    def parent(self):
         return None
 
     def __str__(self):
         return str(self.view.port.name)
+
+
+def _get_root(ref: Ref) -> typing.Optional[Ref]:
+    parent = ref.parent()
+    if parent is None:
+        return None
+    if parent is ref:
+        return ref
+    return _get_root(parent)
```

Comment:
Hmm, since this is called with self, is there a reason it's not just a private instance method?

Reply:
Good point; refactored